assets/src/dashboard/app/views/my-stories/index.js

```diff
@@ -23,37 +23,61 @@ import { __ } from '@wordpress/i18n';
  * External dependencies
  */
 import styled from 'styled-components';
-import { useState } from 'react';
+import { useContext, useEffect, useState } from 'react';
 
 /**
  * Internal dependencies
  */
-import { ViewHeader, FloatingTab } from '../../../components';
-import { storiesFilters } from '../../../constants';
+import {
+  ViewHeader,
+  FloatingTab,
+  CardGrid,
+  CardGridItem,
+  CardTitle,
+} from '../../../components';
+import { STORY_STATUSES } from '../../../constants';
+import { ApiContext } from '../../api/api-provider';
 
 const FilterContainer = styled.div`
   padding: 0 20px 20px;
   border-bottom: ${({ theme }) => theme.subNavigationBar.border};
 `;
 
 function MyStories() {
-  const [currentFilter, setFilter] = useState(storiesFilters[0].value);
+  const [status, setStatus] = useState(STORY_STATUSES[0].value);
+  const { actions } = useContext(ApiContext);
+  const [stories, setStories] = useState([]);
+
+  useEffect(() => {
+    actions.fetchStories({ status }).then(setStories);
+  }, [actions, status]);
+
   return (
     <div>
       <ViewHeader>{__('My Stories', 'web-stories')}</ViewHeader>
       <FilterContainer>
-        {storiesFilters.map((filter) => (
+        {STORY_STATUSES.map((currentStatus) => (
```

Comment:
very much a nit, but `currentStatus` makes me think that it's the _current_  status set on the page, not one of the statuses available through `STORY_STATUSES`, perhaps the singular `storyStatus` would be more of a correlation? Again, real minor - if you feel strongly about the wording then I'm fine with it.